controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -427,7 +432,13 @@ public static int getSegmentCount(final byte[] segmentTable) {
     public static boolean isScaleOngoing(final byte[] historyTable, final byte[] segmentTable) {
 
         HistoryRecord latestHistoryRecord = HistoryRecord.readLatestRecord(historyTable, false).get();
-        return latestHistoryRecord.isPartial() || !latestHistoryRecord.getSegments().contains(getLastSegmentNumber(segmentTable));
+        // Either last row of HistoryTable is partial,
+        // or negation of the following condition.
+        // Non-empty set of segments in last row of HistoryTable implies that
+        // those segments contain highest numbered segment from SegmentTable.
+        return latestHistoryRecord.isPartial() ||
```

Comment:
this check seems to be redundant:

The original said -- either both history and segment tables are updated, in which case history.latest.ispartial == true.
OR only segment table is updated in which case latest segment number should not be in history table. 

Now lets look at new condition added --> 
1. if (history table is updated, and scale is ongoing -- history.latest.ispartial == true. 
2. if only segment table is updated and history table is not updated --
segmentTable.lastSegmentNumber == seal_sentinel.segmentNumber.
historyRecord.ispartial == false --> history.latest will not contain latest segment number. 

So, if the objective is to know if scale is ongoing -- original conditions are enough.
if we want to know if a stream is sealed, history.latest.getSegments = empty should do. 

'!latestHistoryRecord.getSegments().isEmpty() ' is not giving us anything.